src/operator/contrib/bounding_box-inl.cuh

```diff
@@ -280,6 +280,45 @@ void NMSApply(mshadow::Stream<gpu> *s,
   }
 }
 
+__launch_bounds__(512)
+__global__ void nms_calculate_batch_start_kernel(int32_t * batch_start,
+                                                 int32_t * valid_batch_id,
+                                                 size_t N,
+                                                 int num_batch) {
+  size_t tid = blockIdx.x * blockDim.x + threadIdx.x;
+  if (tid < N) {
+    const int32_t previous = tid > 0 ? __ldg(valid_batch_id + tid - 1) : -1;
```

Comment:
Using __ldg intrinsics will fail to compile on some earlier cuda archetectures.

Reply:
It will fail on sm 3.0 and earlier (so Fermi and the first Kepler). I can put ifdef there, but do we care about those?

Reply:
In Makefile, sm 30 is in KNOWN_CUDA_ARCHS.
[https://github.com/apache/incubator-mxnet/blob/master/Makefile#L385](https://github.com/apache/incubator-mxnet/blob/master/Makefile#L385)

Reply:
Then we do ;-). I will introduce the guard, thanks!